awx/main/tests/unit/test_tasks.py

```diff
@@ -328,7 +329,7 @@ def test_with_ssh_credentials(self):
             }
         )
         credential.inputs['password'] = encrypt_field(credential, 'password')
-        self.instance.credential = credential
+        self.instance.credentials.add(credential)
```

Comment:
how on Earth did you get this to work inside of the unit tests?

Reply:
mock hell

Reply:
I tried re-using that approach for the `test_workflow_unit.py` file to mock the m2m relationships. Yeah, okay, I think I can buy that it'll work when there's only ever 1 object involved. But I'm 10 layers deep googling into property mocks, and for the case where you might start with 1 object (with a m2m relationship) and return a 2nd object (with let's say, a copied version of that relationship), I really don't think it's ever going to work..

Thinking for the long-term, I don't see any alternative to the reality that we'll just have to convert a bunch of these unit tests back into functional tests. This will hit hard when the saved launch config stuff comes in. I want @chrismeyersfsu to know this.

Reply:
Yea :/

For what it's worth, there were several tests like this where I really went back and forth on whether I wanted to keep them as unit tests or if I should introduce a real database.